src/main/java/org/truffleruby/aot/ParserCache.java

```diff
@@ -54,12 +54,17 @@ private RootParseNode load(String canonicalPath) {
         final StaticScope staticScope = new StaticScope(StaticScope.Type.LOCAL, null);
         final DynamicScope dynamicScope = new DynamicScope(staticScope);
         final ParserConfiguration parserConfiguration = new ParserConfiguration(null, 0, false, true, false);
+
         final Source source;
+        if (!canonicalPath.startsWith(SourceLoader.RESOURCE_SCHEME)) {
+            throw new UnsupportedOperationException();
```

Comment:
I think adding the bad path to the exception message here would be helpful.

Reply:
Looking now, I see you've just moved the code from elsewhere. Up to you whether to add that now or leave it for another day.

Reply:
Good catch, fixed.